### PR TITLE
Fix bug in jnp.repeat where for array repeats with zero entries (e.g. [1,0,3]) and axis=0 the zero entry was not skipped).

### PR DESCRIPTION
**Pull request by jg8610**

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Looks good!
